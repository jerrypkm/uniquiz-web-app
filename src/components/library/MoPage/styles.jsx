const styles = (theme) => ({
  section: {
    maxHeight: "100%",
    height: "100%",
    width: "100%",
    paddingTop: theme.space?.lg,
    paddingBottom: theme.space?.lg,
    [theme.breakpoints.up(400)]: {
      paddingTop: theme.space?.lg,
      paddingBottom: theme.space?.lg,
    },
  },
  textAlignRight: { textAlign: "right" },
});

export default styles;
