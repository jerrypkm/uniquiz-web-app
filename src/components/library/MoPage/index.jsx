/**
 * MoPage
 * Default page used through out the app, used to wrap elements with page like styles
 * provides a heading title and a subtitle for the page with loading spinner
 * @param {Boolean} isAdmin - based on user role
 * @param {String} title - Title for the page
 * @param {String} subtitle - Subtitle for the page
 */

import React from "react";

import MoTypography from "../MoTypography";
import styles from "./styles";
import withStyles from "@material-ui/core/styles/withStyles";

const MoPage = ({ classes, children, IconComponent, subtitle, title }) => {
  return (
    <section className={classes.section}>
      <MoTypography
        font="breeSerif"
        marginBottom="md"
        text={title}
        variant="h2"
      >
        {IconComponent && <IconComponent />}
      </MoTypography>
      <MoTypography
        font="openSans"
        marginBottom="md"
        text={subtitle}
        variant="h6"
      ></MoTypography>
      {children}
    </section>
  );
};

export default withStyles(styles)(MoPage);
