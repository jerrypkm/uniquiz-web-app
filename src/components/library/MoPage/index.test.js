import React from "react";

import ShallowRenderer from "react-test-renderer/shallow";
import MoPage from "./index.jsx";

const renderer = new ShallowRenderer();

describe("MoPage Component", () => {
  test("should match the snapshot", () => {
    renderer.render(<MoPage />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
