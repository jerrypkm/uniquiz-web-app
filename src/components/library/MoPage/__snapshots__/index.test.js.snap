// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoPage Component should match the snapshot 1`] = `
<MoPage
  classes={
    Object {
      "section": "MoPage-section-1",
      "textAlignRight": "MoPage-textAlignRight-2",
    }
  }
/>
`;
